src/components/kick-button.js

```diff
@@ -0,0 +1,24 @@
+AFRAME.registerComponent("kick-button", {
+  init() {
+    this.onClick = () => {
+      this.kick(this.owner);
+    };
+    NAF.utils.getNetworkedEntity(this.el).then(networkedEl => {
+      this.owner = networkedEl.components.networked.data.owner;
+    });
+  },
+
+  play() {
+    this.el.addEventListener("grab-start", this.onClick);
+  },
+
+  pause() {
+    this.el.removeEventListener("grab-start", this.onClick);
+  },
+
+  async kick(clientId) {
+    const { permsToken } = await this.el.sceneEl.systems.permissions.fetchPermissions();
+    NAF.connection.adapter.kick(clientId, permsToken);
+    window.APP.hubChannel.kick(clientId);
```

Comment:
Note I've changed the NAF `clientId`, it is now just the phoenix socket `session_id`. Not sure if that's a bad idea, but I couldn't think of a better way of correlating a NAF client to a phoenix socket.

Reply:
that sounds right to me